### PR TITLE
[GraphBolt] remove pydantic_yaml from dependency

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

fix for https://github.com/dmlc/dgl/issues/6010.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
@dgl-bot

File: tests/python/pytorch/graphbolt/test_ondisk_dataset.py [hunk 1]
```diff
@@ -637,7 +637,7 @@ def test_OnDiskDataset_Graph_Exceptions():
 
         with pytest.raises(
             pydantic.ValidationError,
-            match="value is not a valid enumeration member",
+            match="1 validation error for OnDiskMetaData",
```
[the remark this answers is not in the document]
Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/test_ondisk_dataset.py:640]: the number of validation error

[the remark this answers is not in the document]
Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/test_ondisk_dataset.py:640]: Acutally, it's better to capture `value is not a valid enumeration member` but I found such error log is printed out in windows only, On linux, such error message is not printed out. Probably caused by different pydantic version or different implementation of pydantic on different platforms.

File: tests/python/common/data/test_data.py [hunk 2]
```diff
@@ -1668,6 +1668,9 @@ def _test_NodeEdgeGraphData():
     reason="Datasets don't need to be tested on GPU.",
 )
 @unittest.skipIf(dgl.backend.backend_name == "mxnet", reason="Skip MXNet")
+@unittest.skipIf(
```
Comment (Rhett-Ying) [inline tests/python/common/data/test_data.py:1671]: due to `AttributeError: module 'typing' has no attribute 'OrderedDict'`. seems need to upgrade python version

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**